### DISCUSSION TITLE
Wave share DSI 7.9” LCD 1280 x 400

### DISCUSSION THREAD

**Discussion by Wandereradi** [rern/rAudio#1482] [Feature Requests]
Can the above display be used with raudio. 

Would love to have the support for this display 

https://www.waveshare.com/wiki/7.9inch_DSI_LCD#Method_2:_Rotation_Display_In_Lite_Version

**Comment by cYCL157** [comment]
If you can enable via shell script, you might be able to use a custom startup/shutdown script.  That was a recent answer to another question:

https://github.com/rern/rAudio/discussions/1466#discussioncomment-7360779

**Comment by Wandereradi** [comment]
if SSH was transparent i could try to run the command for driver
installations.

**Comment by lovehifi** [comment]
@Wandereradi See the topic:

[Waveshare DSI LCD displays](https://community.volumio.com/t/waveshare-dsi-lcd-displays/57991)

**Comment by Wandereradi** [comment]
Hey Lovehifi,

to apply for the driver, I need access to ssh. I need ssh to be able to use
cd and other commands that are locked in r  audio. any tips for that.

**Reply by ceejayemm** [reply]
Create a blank file called 'ssh' (no quotes and no extension).  Place it in the boot folder/partition of the SD card and reboot your RPi.  This will enable SSH on rAudio and you can login with user 'root' and (default) password 'ros' (again no quotes).  You will then have full access to the running RPi file system and you can do whatever you wish - but BEWARE YOU CAN ANYTHING INCLUDING DESTROYING YOUR INSTALLATION !!!

Chris

**Reply by Wandereradi** [reply]
well i tried it but it opens the restricted ssh window , where many of the commands do not work. so it does not seem possible for a noob like me install the driver for 7.9" waveshare dsi display. 

this is the same situation like in Moode audio.

the only place we have complete access is volumio but that has its own chalanges with wifi reconnection.

**Reply by ceejayemm** [reply]
Works for me every time. What does:

ls - lrt /boot/ssh

show ?

Are you connecting from another PC using:

ssh root@xxx.xxx.xxx.xxx

Where xxx.xxx.xxx.xxx is the ip number of the RPi running rAudio?

**Comment by Wandereradi** [comment]
Hey Chris,

Thanks for the tip! I will report back if the display worked.

thank you

Aditya

**Comment by Wandereradi** [comment]
[image: image.png]

winscp
[image: image.png]
the above right?  as per your direction or did miss something

regards
Aditya

**Reply by ceejayemm** [reply]
Sorry, the image you posted didn't show up so I can't see what you were doing with WinSCP.

Lets take things back a little.  How did you originally write the image to your SD card ?  Via Raspberry Imager or Belena.io on a PC/Mac?  If so, power down the RPi running rAudio. Take out the SD card and put it back into your PC/Mac.  This should open 2 partitions in the file manger - ROOT and BOOT.  Create a new, empty file called ssh and copy this file to the BOOT partition (the same partition which should already contain the config.txt file. Eject the SD card from the PC/Mac and use it to reboot the RPi.  Once the RPi is reboot,from the command line / terminal on the PC/Mac, start an SSH session using:

ssh root@xxx.xxx.xxx.xxx

where xxx.xxx.xxx.xxx is the IP number of your RPi.  You might need to accept the new fingerprint of the RPI, type YES (might be case sesnitive) to do this. When prompted for a password enter 'ros' (without the quotes) and you should now have access to to the RPi root user folder with full access to the entire RPi filesystem.  As before BE CAREFUL as you can destroy your system if you are not sure what you are doing.

Chris

**Comment by Wandereradi** [comment]
Hi Chris,

I used the tool Balena etcher to write the image of rAudio to the sd card.
Then I put the card in the PI and run it to set up the wifi and the DAC.
After which I used winscp to create a file in the boot folder called ssh
and after this shut down the pi using the web interface and cycle back the
power to turn the pi on.

After the above is done, I use putty for ssh, I am able to use the "root"
as user id and "ros" as the password but after it go forward the next
interface on putty is custom which does not allow cd or dir hence i am
unable to install the driver from waveshare.

Let me know if I am going wrong anywhere.

attached the images here 

![raudio putty](https://github.com/rern/rAudio/assets/127820040/2bac37b1-fdbe-42b7-8ae2-9754ebbdfd32)
![raudio winscp](https://github.com/rern/rAudio/assets/127820040/3b508113-0e53-4970-9745-1cf4a1078766)


Regards
Aditya

**Reply by ceejayemm** [reply]
So Putty is getting you connected to the rAudio file system as you are being shown the rAudio:~ prompt (first image).  I guess you are new to Linux from what this image shows. I would suggest getting to know Linux a bit better before you try to install the display driver.  For instance, the 'cd' command should be cd .. (with a space between cd and the two dots) to move up a folder level hence the error you are getting.  However by logging in as root you are already at the top folder level so it can't go any higher.  Try ls / and/or ls -lrt / (again with a space between the command and the arguments) to have a look at the folders you have available, then cd /tmp will take you to the lower level tmp folder for example.

I suspect that Linux might be new to you so,as I have said several times before, please take care as, if you do something you are not sure of, this can have catastrophic consequences. If you do damage the installation you will have no option but to reflash the SD card and start again.  I am afraid I am not in a position to teach you Linux - sorry.

Chris

**Comment by Wandereradi** [comment]
Hi Chris,

I have managed to install the same driver to Volumio, and did not face
issue.

The issue with Volumio is the WiFi always acts funny after a shutdown cycle
and out of 5 it will fails 2 times

And the other mistake ( if I can call it so ) is that I bought the DSI
display but now I realize I should have bought the HDMI display

Thank for the patience with me.

You are right I am noob 💡 to Linux.

Regards
Aditya

**Comment by SO3D** [comment]
Hi,
I also have a screen as in the subject.

I have the latest image, I added the following configuration to the file
/boot/config.txt

_hdmi_group=2
hdmi_mode=87
hdmi_timings=400 0 100 10 140 1280 10 20 20 2 0 0 0 60 0 43000000 3
display_rotate=3_

The image on the screen displays correctly, but the touch does not work. How to fix it?
